### PR TITLE
Fix InputEventScreenDrag on Android

### LINKED ISSUE #84138: Touch tracking on Android cannot register an initial movement less than many pixels
### Godot version

v4.1.2.stable

### System information

Android 11

### Issue description

On Android neither InputEventScreenDrag, InputEventMouseMove or get_global_mouse_position() can detect the movement of a touch until it passes a pixel threshold (20+ pixels on minimal project included on my phone)

I'm attempting to have fine control of a screen element on mobile using touch. Specifically for controlling a pick in a lock picking mini-game. Current testing is on Android.

But regardless of the input methods I've tried after pressing on the pick there is no motion until the pick suddenly catches up with the touch once it has moved 20+ pixels from the starting location.

I appreciate this is standard behaviour on drag for a desktop UI but for a game it's not desirable so I went hunting for a setting either globally or specific to the scene that controls the drag threshold and found none. 

I then tried reimplementing using _input instead of _gui_input, I tried responding to InputEventMouseMove directly rather than the drag event. Eventually I put a call to get_global_mouse_position() in the process() function. But all show exactly the same behaviour. Although the InputEventScreenTouch event gives a precise location on touch and release, no Move or Drag events occur until the touch is moved by many pixels. Even get_global_mouse_position() shows no change in position until the touch has moved the same number of pixels.

This seems to mean fine touch control of a screen element is impossible on Android at the moment with Godot 4.
I'm hoping I'm missing something.

I can see there have been previous issues with people talking about a drag "lag" but this seems to be a different issue.

### Steps to reproduce

In the minimal reproducible project below, run the project on desktop, click on the white square and move the mouse.
The labels will register the relative movement of the mouse when movement was detected either using motion events, drag events or get_global_mouse_position(). On desktop this will immediate and can show motion of +/- 1 pixel.

Run the same project on Android and the labels will not register any motion until the touch has moved a significant distance.

### Minimal reproduction project

[BugReport.zip](https://github.com/godotengine/godot/files/13197907/BugReport.zip)

### LINKED ISSUE #84182: get_global_mouse_position() updates delayed after initial tap
### Godot version

v4.1.2.stable.official [399c9dc39], v4.2.beta3.official [e8d57afae]

### System information

Android, Galaxy Tab S6 Lite

### Issue description

Hey all,

I have encountered a weird bug on android. I want to make a handwriting app. But when it comes to drawing, weird stuff happens:

When touching and drawing with the pen, `event.position`, `get_global_mouse_position()` and `get_viewport().get_mouse_position()` return the correct position of the event, but while dragging, the next position has a 12px position from the initial tap. After that, the position updates just fine. This is't ideal for handwriting obviously.

This happens in godot 4.1.2 and 4.2 beta (using the beta, because some tablet stuff got optimised there)

The Tablet is a Samsung Galaxy Tab S6 Lite and it juses the non-bluetooth S-Pen (I'm affraid that it could be caused by the non active - active Pen type)

I don't have any additional devices for testing though.

### Steps to reproduce

- Just create a new Project on either 4.2 or 4.1.2
- Create Node2D and attach script below
- Debug on Android (won't work for 4.2 because the android debugger is not working currently see [this Issue](https://github.com/godotengine/godot/issues/82764#issue-1925350903)

### Minimal reproduction project

This is the Script that shows the problem. You can see the 12.xx px distance from the first position there. (zipped project bigger than 25MB)

```
extends Node2D
var activeLine = null
var dist_to_last: float
var last_mouse_pos = Vector2(0,0)
var current_mouse_pos = Vector2(0,0)

func _process(delta):
	if Input.is_action_pressed("mouse_click"):
		current_mouse_pos = get_global_mouse_position()
		dist_to_last = (current_mouse_pos - last_mouse_pos).length()
		if dist_to_last > 1:
			print(dist_to_last)
		last_mouse_pos = current_mouse_pos
```

### PR DESCRIPTION
**Pull request by Alex2782**
Fixes: #84138 
Fixes: #84182

I haven't checked the effects on other events yet.
Event `onScroll` is not fired directly and is not so good for 'ScreenDrag': 
<details><summary>androidx.core sources</summary>

image
image
</details>

**Tests**

[test-project.zip](https://github.com/godotengine/godot/files/13232340/test-project.zip)

v4.2.beta2.official [f8818f85e]

The "Drag At" and "Motion At" values are higher (the fingers have to be moved more)

https://github.com/godotengine/godot/assets/41921395/87b3e269-cd65-4cb7-babb-5be21a616f3e

**this PR:**

https://github.com/godotengine/godot/assets/41921395/5ece2845-7bad-4eaf-83fc-9d3aea736cb2

### CODE REVIEW

**Review by m4gr3d — APPROVED** [review]
The logic looks good.

Can you add comments to the added code to clarify what it does.

**Review by m4gr3d — CHANGES_REQUESTED** [review]
I'm seeing some scrolling artifacts when testing these changes using the Android editor; can you address them.


https://github.com/godotengine/godot/assets/914968/03bf53a9-bd3a-4e09-8720-4320e54209f5

**Comment by Alex2782** [comment]
I'll try to find time for it again in 2 or 3 days.

**Comment by Alex2782** [comment]
@m4gr3d The new solution should be better, easier to understand and probably also perform better.  I tried out various gestures with Godot Editor and checked them in Logcat. Checking 'pointerCount' is not necessary as it was in my old version.

**Comment by akien-mga** [comment]
@Alex2782 Did you intend to close this? Or did you delete the branch by mistake while cleaning up?

**Comment by Alex2782** [comment]
sorry, my mistake, I didn't think it would be deleted here too.

**Comment by m4gr3d** [comment]
@Alex2782 Your commit still had a few issues that I addressed in db6f1c9e2f36721d7a7f7fac17f4baebdb31615d. Can you take a look and merge the commit together if the updates look good.

**Review by m4gr3d — APPROVED** [review]
Approved with the updates included in db6f1c9e2f36721d7a7f7fac17f4baebdb31615d.

The commits will need to be squashed together prior to merging.

File: platform/android/java/lib/src/org/godotengine/godot/input/GodotGestureHandler.kt [hunk 1]
```diff
@@ -189,6 +194,17 @@ internal class GodotGestureHandler : SimpleOnGestureListener(), OnScaleGestureLi
 				sourceMouseRelative
 			)
 			return true
+		} else if (!scaleInProgress) {
+			// The 'onScroll' event is triggered with a long delay.
+			// Force the 'InputEventScreenDrag' event earlier here.
+			// We don't toggle 'dragInProgress' here so that the scaling logic can override the drag operation if needed.
+			// Once the 'onScroll' event kicks-in, 'dragInProgress' will be properly set.
+			if(lastDragX != event.getX(0) || lastDragY != event.getY(0)) {
```
Comment (akien-mga) [inline platform/android/java/lib/src/org/godotengine/godot/input/GodotGestureHandler.kt:202]:
```suggestion
			if (lastDragX != event.getX(0) || lastDragY != event.getY(0)) {
```

**Comment by m4gr3d** [comment]
@Alex2782 Looks like the rebase wasn't done properly.

**Comment by Alex2782** [comment]
Maybe now.

It still works like in the video:

https://github.com/godotengine/godot/assets/41921395/5ece2845-7bad-4eaf-83fc-9d3aea736cb2

**Review by m4gr3d — APPROVED** [review]
Looks good!

**Review by m4gr3d — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**